compiler/AST/iterator.cpp

```diff
@@ -493,7 +511,8 @@ bool checkIteratorFromForExpr(Expr* ref, Symbol* shape) {
 // if the field did not exist.
 //
 CallExpr* setIteratorRecordShape(Expr* ref, Symbol* ir, Symbol* shapeSpec,
-                                 bool fromForExpr) {
+                                 bool fromForExpr,
+                                 bool fromForeachExpr) {
```

Comment:
This is where flags are for `for`/`foreach`, where above they were for `foreach`/`forall`. Unifying these two with an enum would be a really worthy improvement.